pkg/types/types.go

```diff
@@ -72,6 +73,14 @@ type Address []byte
 // upkeep at a block number
 type UpkeepKey []byte
 
+func (u UpkeepKey) BlockKeyAndUpkeepID() (blockKey string, upkeepID string) {
```

Comment:
This function should not return strings. The return types should be `BlockKey` and `UpkeepIdentifier`

Reply:
Okay cool; a lot of the client code that uses this functionality expects these values as strings, are we okay with casting the BlockKey and UpkeepIdentifier any time they're needed?

Reply:
I think that is fine. I would prefer using distinct types for BlockKey and UpkeepIdentifier because functions that should expect them as inputs will eventually benefit from the type references as each morphs into interfaces. Casting should be fine until we create helper functions such as `BlockKey.After(BlockKey)`. These will likely be needed later.